### PR TITLE
feat: send notification using notification serivce

### PR DESCRIPTION
**Pull request by kaladivo**

### CODE REVIEW

**Comment by kaladivo** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/1032?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1045** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/1045?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1032** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/1032?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#954** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/954?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kaladivo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/resources-utils/src/notifications/callWithNotificationService.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      SemverString.compare(otherSideVersion)(
+        '<',
+        FE_VERSION_SUPPORTING_V2_NOTIFICATIONS
+      )
+    )
+      return f({...(fArgs as T), notificationServiceReady: false})
+
+    return pipe(
+      f({...(fArgs as T), notificationServiceReady: true}),
+      TE.chainW((result) => {
```
Comment (SamTremko) [inline packages/resources-utils/src/notifications/callWithNotificationService.ts:51]: pls make this more readable

File: packages/resources-utils/src/notifications/notificationPayloadCrypto.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+} from '@vexl-next/cryptography/src/KeyHolder'
+import {
+  ChatNotificationData,
+  EncryptedNotificationPayload,
+} from '@vexl-next/domain/src/general/notifications'
+import {toError} from '@vexl-next/domain/src/utility/errors'
+import {Effect} from 'effect'
+import {type CryptoError} from '../utils/crypto'
+
+export class EncryptionError extends Schema.TaggedError<EncryptionError>()(
```
Comment (SamTremko) [inline packages/resources-utils/src/notifications/notificationPayloadCrypto.ts:16]: pls remove if not used

File: packages/resources-utils/src/notifications/notificationPayloadCrypto.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    ),
+    Effect.map(EncryptedNotificationPayload)
+  )
+}
+
+export function decryptChatNotificationPayload(
+  privateKey: PrivateKeyPemBase64
+): (
+  encrypted: EncryptedNotificationPayload
+) => Effect.Effect<ChatNotificationData, CryptoError | ParseResult.ParseError> {
```
Comment (SamTremko) [inline packages/resources-utils/src/notifications/notificationPayloadCrypto.ts:49]: Unify ParseError

File: eas.json
Comment (SamTremko) [inline eas.json:1]: pls check is removed

File: packages/resources-utils/src/chat/sendMessagingRequest.ts [hunk 4]
```diff
@@ -63,6 +86,9 @@ export function sendMessagingRequest({
   lastReceivedFcmCypher?: FcmCypher
   api: ChatPrivateApi
   myVersion: SemverString
+  theirFcmCypher?: FcmCypher
```
Comment (SamTremko) [inline packages/resources-utils/src/chat/sendMessagingRequest.ts]: pls add | undefined

File: packages/resources-utils/src/chat/sendMessage.ts [hunk 5]
```diff
@@ -20,16 +27,40 @@ export type SendMessageApiErrors = ExtractLeftTE<
...
   senderKeypair,
+  theirFcmCypher,
+  notificationApi,
+  otherSideVersion,
 }: {
   api: ChatPrivateApi
   receiverPublicKey: PublicKeyPemBase64
   message: ChatMessage
   senderKeypair: PrivateKeyHolder
+  theirFcmCypher?: FcmCypher
```
Comment (SamTremko) [inline packages/resources-utils/src/chat/sendMessage.ts]: | undefined

File: packages/domain/src/general/notifications.ts [hunk 6]
```diff
@@ -31,3 +43,50 @@ export function extractCypherFromFcmCypher(
 
   return split
 }
+
+export const ChatNotificationType = S.Literal(
+  'MESSAGE',
```
Comment (SamTremko) [inline packages/domain/src/general/notifications.ts:48]: pls check TradeChecklist notification

**Review by SamTremko — APPROVED** [review]

**Pull request closed without merging**